File: EASSET.Web/Modules/MasterData/Ruang/RuangForm.cs
using Serenity.ComponentModel;
using System;

namespace EASSET.MasterData.Forms;

[FormScript("MasterData.Ruang")]
[BasedOnRow(typeof(RuangRow), CheckNames = true)]
public class RuangForm
{
    public string RuangKode { get; set; }
    public string RuangNama { get; set; }
    public string UnitId { get; set; }
    public string SubUnitId { get; set; }
    public string ThAng { get; set; }
    //public string Op { get; set; }
    //public DateTime Lu { get; set; }
    //public string Pc { get; set; }
    //public short Dlt { get; set; }
}

File: EASSET.Web/Modules/ServerTypes/MasterData/RuangForm.ts
import { PrefixedContext, initFormType } from "@serenity-is/corelib";
import { StringEditor, ServiceLookupEditor } from "serenity.corelib";

export interface RuangForm {
    RuangKode: StringEditor;
    RuangNama: StringEditor;
    UnitId: ServiceLookupEditor;
    SubUnitId: ServiceLookupEditor;
    ThAng: StringEditor;
}

export class RuangForm extends PrefixedContext {
    static readonly formKey = 'MasterData.Ruang';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!RuangForm.init)  {
            RuangForm.init = true;

            var w0 = StringEditor;
            var w1 = ServiceLookupEditor;

            initFormType(RuangForm, [
                'RuangKode', w0,
                'RuangNama', w0,
                'UnitId', w1,
                'SubUnitId', w1,
                'ThAng', w0
            ]);
        }
    }
}